### PR TITLE
Add partial_fit in Potato

### PR DESCRIPTION
**Pull request by qbarthelemy**
This PR:
- add function `partial_fit()` in `Potato`, useful for online update;
- complete tests;
- update example on artifact detection.

### CODE REVIEW

**Comment by qbarthelemy** [comment]
@lkorczowski  has put the artifact detection example in a Python notebook here:
https://github.com/lkorczowski/BCI-2021-Riemannian-Geometry-workshop/blob/master/notebooks/plot_detect_riemannian_potato_EEG.ipynb
with a nice javascript animation, adding these lines at the end
```
from IPython.display import HTML
plt.rcParams["animation.html"] = "jshtml"
potato
```

Following this idea, I have added these lines:
```
from IPython.display import HTML
plt.rcParams["animation.embed_limit"] = 10
HTML(potato.to_jshtml(fps=5, default_mode='loop'))
```
But, in the new documentation, the last figure is plotted two times: with `plt.show()` and then with `HTML(potato.to_jshtml())`.
I don't know how to detect when code is executed in a ipynb, and in this case, do not call `plt.show()`. Any idea?

File: doc/whatsnew.rst [hunk 1]
```diff
@@ -10,9 +10,11 @@ A catalog of new features, improvements, and bug-fixes in each release.
 v0.2.8.dev
 ----------
 
+- Correct spectral estimation in :func:`pyriemann.utils.covariance.cross_spectrum` to obtain equivalence with SciPy
+
 - Add instantaneous, lagged and imaginary coherences in :func:`pyriemann.utils.covariance.coherence` and :class:`pyriemann.estimation.Coherences`
 
-- Correct spectral estimation in :func:`pyriemann.utils.covariance.cross_spectrum` to obtain equivalence with SciPy
+- Add `partial_fit` in :class:`pyriemann.clustering.Potato`, useful for an online update; and update example on artifact detection
```
Comment (agramfort) [inline doc/whatsnew.rst:17]:
```suggestion
- Add ``partial_fit`` in :class:`pyriemann.clustering.Potato`, useful for an online update; and update example on artifact detection.
```

**Review by agramfort — APPROVED** [review]
besides LGTM

**Review by agramfort — APPROVED** [review]
merge if happy @qbarthelemy

**Review by sylvchev — APPROVED** [review]
All good to me!

**Comment by sylvchev** [comment]
> I don't know how to detect when code is executed in a ipynb, and in this case, do not call `plt.show()`. Any idea?

This could be useful: https://stackoverflow.com/a/22424821

**Comment by agramfort** [comment]
@lkorczowski green button is yours

**Pull request merged**